Integrating T-Mobile API

svetlana opened the pull request:
@blatyo -- Allen, I think I'm fully done with WIT changes. The only thing I am planning for WIT is to squash all of these commits right before the merge.

blatyo commented on Gemfile:
```diff
@@ -10,8 +10,12 @@ gem 'newrelic_rpm'
 gem 'foreman'
 gem 'gz_release', require: false
+gem 'sentry-raven'
+gem 'httparty'
```
doesn't look like httparty is getting used

blatyo commented on lib/esn.rb:
```diff
@@ -8,10 +9,12 @@
     case carrier.to_s.downcase
     when 'tmobile'
-      ESN::TMobile
+      ENV['USE_TMOBILE_API'] =~ /^true$/i ? ESN::TMobile::Api : ESN::TMobileSiteScraper
```
Since SiteScraper doesn't work. We probably don't need this and you can delete the site scraper code.

blatyo commented on spec/esn/tmobile/api_spec.rb:
```diff
@@ -0,0 +1,141 @@
+      ENV['RACK_ENV'] = 'TESTING'
+      ENV['USE_TMOBILE_API'] = 'true'
+                  
+      ESN::TMobile::Api::GEN_TOKEN_AUTH = 'Basic fake_oauth_token'
```
Instead of redefining constants, you can use dotenv-rails to load environment variables with these values in a `.env.test` file. It also allows you to create a `.env.local` file to override those, when you want to test real values.

svetlana commented:
@blatyo -- ready for another review.

1. Removed Site Scraper.
2. Moved most functionality from Base to Verizon.
3. Attempted to use dotenv for loading test configs, but it was not pre-loading them, and I am not sure why not.
4. Could you take a look at these warnings please? https://circleci.com/gh/secondrotation/where_is_tudor/115 (in rspec logging)
